src/KMonad/App/Main.hs

```diff
@@ -84,16 +84,19 @@ initAppEnv cfg = do
   src <- using $ cfg^.keySourceDev
 
   -- Initialize the pull-chain components
-  dsp <- Dp.mkDispatch $ awaitKey src
-  ihk <- Hs.mkHooks    $ Dp.pull  dsp
-  slc <- Sl.mkSluice   $ Hs.pull  ihk
+  itr <- atomically $ newEmptyTMVar
```

Comment:
Apply hint
```suggestion
  itr <- newEmptyTMVarIO
```
This is a problem at a few points.
Even better rebase on `master` since their it's already fixed.